iceoryx_posh/test/mocks/client_mock.hpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_MOCKS_CLIENT_MOCK_HPP
+#define IOX_POSH_MOCKS_CLIENT_MOCK_HPP
+
+#include "iceoryx_hoofs/cxx/expected.hpp"
+#include "iceoryx_posh/internal/popo/base_client.hpp"
+#include "mocks/base_port_mock.hpp"
+
+#include "test.hpp"
+
+using namespace ::testing;
+using ::testing::_;
+
+class MockClientPortUser : public MockBasePort
+{
+  public:
+    using MemberType_t = iox::popo::ClientPortData;
+    MockClientPortUser(MemberType_t&){};
+    MockClientPortUser() = default;
+
+    MockClientPortUser(const MockClientPortUser&) = delete;
+    MockClientPortUser& operator=(const MockClientPortUser&) = delete;
+    MockClientPortUser(MockClientPortUser&&) noexcept
+    {
+    }
+    MockClientPortUser& operator=(MockClientPortUser&&) noexcept
+    {
+        return *this;
+    }
```

Comment:
Rule of 5 is violated, please add `~MockClientPortUser() = default`.  (clang-tidy warning)